src/main/java/ch/njol/skript/expressions/ExprNewBannerPattern.java

```diff
@@ -0,0 +1,133 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.bukkitutil.BukkitUtils;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.banner.Pattern;
+import org.bukkit.block.banner.PatternType;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Stream;
+
+@Name("New Banner Pattern")
+@Description("Create a new banner pattern.")
+@Examples({
+	"set {_pattern} to a new creeper banner pattern colored red",
+	"add {_pattern} to banner patterns of {_banneritem}",
+	"remove {_pattern} from banner patterns of {_banneritem}",
+	"set the 1st banner pattern of block at location(0,0,0) to {_pattern}",
+	"clear the 1st banner pattern of block at location(0,0,0)"
+})
+@Since("INSERT VERSION")
+public class ExprNewBannerPattern extends SimpleExpression<Pattern> {
+
+	private static Object[] patternTypes;
+	private static Map<Integer, Object> patternCorrelation = new HashMap<>();
+
+	static {
+		if (BukkitUtils.registryExists("BANNER_PATTERN")) {
+			try {
+				Class<?> registryClass = Class.forName("org.bukkit.Registry");
+				Object bannerRegistry = registryClass.getField("BANNER_PATTERN").get(null);
+				List<?> registryPatterns = ((Stream<?>) registryClass.getMethod("stream").invoke(bannerRegistry)).toList();
+				patternTypes = registryPatterns.toArray();
+			} catch (ClassNotFoundException | NoSuchFieldException | IllegalAccessException |
+					 InvocationTargetException | NoSuchMethodException e)
+			{
+				throw new RuntimeException(e);
+			}
+
+		} else {
+			try {
+				Class<?> enumClass = Class.forName("org.bukkit.block.banner.PatternType");
+				if (enumClass.isEnum()) {
+					patternTypes = enumClass.getEnumConstants();
+				} else {
+					throw new IllegalStateException("PatternType is neither an enum nor a valid registry.");
+				}
+			} catch (ClassNotFoundException | IllegalStateException e) {
+                throw new RuntimeException(e);
+            }
+        }
+		List<String> patterns = new ArrayList<>();
+		for (int i = 0; i < patternTypes.length; i++) {
+			Object type = patternTypes[i];
+			try {
+				patterns.add("[a] [new] "
+					+ getKey(type)
```

Comment:
why not just register a new type for the patterns?

Reply:
Do you think this would  be better? I assumed having all this mess with reflection and stuff would be better to have in this class rather than in the BukkitClasses

Reply:
well i would make a banner module honestly
but yes it would be way better

Reply:
The only problem I see by adding the type, is that it may conflict with the `banner patterns` within skript-aliases. If the patterns I set within the lang consist of the words `banner pattern` such as `creeper: creeper banner pattern`

Reply:
try it and see

Reply:
So it did end up working, but then it would be a breaking change because, even though this does work, 
```!give me creeper banner pattern```
Doing this doesnt work
```
!set {item} to creeper banner pattern
!give me {item}
```
To me it just seems like a discrepancy by having them added as a class info, especially if I do the suggestion of putting `%*color%` in front.
People might think doing `red creeper  banner pattern` would give an item as the existence of banner patterns is already perceived as that.
Thoughts?

Reply:
Unless I make a new expression doing `ExprBannerItem`
that uses the now made banner type class info and gets the corresponding banner item, like `%bannerpattern% item`  or something

Reply:
My concerns mainly lie in that the pattern names will be at the whims of the keys/names of the patterns in the registry.
With a type, we can set the various names in the lang files.

Reply:
That is true, so how should we or I approach this?

Reply:
well there's precedent here with stuff like `chicken` and `a chicken`
since you can do `1 of creeper banner pattern` to avoid the ambiguity i'm willing to accept the breaking change
idk about the others' opinions.

Reply:
alternatively, we could get wacky and use unique names, like `stamp` instead of `pattern`, or something similarly different.

Reply:
Im down with whatever, what about `creeper pattern type`?

Reply:
I'm not a fan of just appending type to things, as it doesn't really mean much in general and is already confusing enough with enchantment vs enchantment type and same with potion effects. I'm in favor of as-is, with the breaking change, tbh.